internal/net/grpc/client.go

```diff
@@ -155,7 +155,7 @@ func (g *gRPCClient) StartConnectionMonitor(ctx context.Context) (<-chan error,
 	}
 	g.monitorRunning.Store(true)
 
-	addrs := make([]string, len(g.addrs))
+	addrs := make([]string, 0, len(g.addrs))
 	for addr := range g.addrs {
 		addrs = append(addrs, addr)
 	}
```

Comment:
Thank you for your contribution and reporting a bug.

How about using slices/maps package to get key addr slice.
FYI: 
- https://pkg.go.dev/slices#Collect
- https://pkg.go.dev/maps#Keys

code changes like below
```suggestion
        addrs := slices.Collect(maps.Keys(g.addrs))
```

and needs import.

Reply:
My pleasure.